app/routers/weekly_tasks.py

```diff
@@ -0,0 +1,256 @@
+import calendar
+import datetime
+
+from typing import List, Tuple, Optional
+from fastapi import APIRouter, Cookie, Depends, Request, Response, Form
+from sqlalchemy.orm.session import Session
+from starlette.responses import RedirectResponse, HTMLResponse
+from starlette.status import HTTP_302_FOUND, HTTP_303_SEE_OTHER
+
+from app.database.models import WeeklyTask
+from app.dependencies import get_db, templates
+from app.internal.security.dependencies import (
+    current_user,
+    schema,
+    is_logged_in,
+    current_user_from_db,
+)
+from app.internal.weekly_tasks import (
+    remove_weekly_task,
+    weekly_task_from_input,
+    create_weekly_task,
+    change_weekly_task,
+)
+
+
+router = APIRouter(
+    prefix="/weekly-tasks",
+    tags=["weekly-tasks"],
+    responses={404: {"description": "Not found"}},
+    dependencies=[Depends(get_db)],
+)
+
+
+def get_checked_days(days: str = "") -> List[Tuple[str, str, str]]:
+    """Produces the input checked_days for the template add_edit_weekly_task"""
+    days_list = days.split(", ")
+    checked_days = []
+    for day_full_name in calendar.day_name:
+        day = day_full_name[:3]
+        day_lower = day.lower()
+        if day in days_list:
+            checked_days.append((day_full_name, day_lower, "checked"))
+        else:
+            checked_days.append((day_full_name, day_lower, ""))
+    return checked_days
+
+
+def get_days_string(
+    sun: bool,
+    mon: bool,
+    tue: bool,
+    wed: bool,
+    thu: bool,
+    fri: bool,
+    sat: bool,
+) -> str:
+    """Produces a string of all the days that were checked,
+    For use in the model weekly tasks."""
+    days_dict = {
+        "Mon": mon,
+        "Tue": tue,
+        "Wed": wed,
+        "Thu": thu,
+        "Fri": fri,
+        "Sat": sat,
+        "Sun": sun,
+    }
+    days_list = [day for day, is_checked in days_dict.items() if is_checked]
+    days = ", ".join(days_list)
+    return days
+
+
+@router.get("/")
+async def weekly_tasks_manager(
+    request: Request,
+    user: schema.CurrentUser = Depends(current_user_from_db),
+):
+
+    # TODO: Move the below function to a compatible location
+    # Need to run regularly whenever there are tasks on the week
+    # Or will run on the background after the user left the
+    # weekly-tasks manager page
+    # function:
+    # generate_tasks(user, session)  # imported from app.internal.weekly_tasks
+    # session.close()
+
+    return templates.TemplateResponse(
+        "weekly_tasks_manager.html",
+        {"request": request, "weekly_tasks": user.weekly_tasks},
+    )
+
+
+@router.get("/add", dependencies=[Depends(is_logged_in)])
```

Comment:
Add `is_logged_in` to the router's dependencies instead

Reply:
I realized that the function is meant to be used in this way as well.
In case I do not need to use the variable it return's inside the router.